instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderApplicationListener.java

```diff
@@ -53,35 +54,126 @@ private static boolean isAssignableFrom(Class<?> type, Class<?>... supportedType
   @Override
   public void onApplicationEvent(ApplicationEvent event) {
     if (event instanceof ApplicationEnvironmentPreparedEvent // Event for which
-        // org.springframework.boot.context.logging.LoggingApplicationListener
-        // initializes logging
-        && !isOpenTelemetryAppenderAlreadyConfigured()) {
-      ch.qos.logback.classic.Logger logger =
-          (ch.qos.logback.classic.Logger)
-              LoggerFactory.getILoggerFactory().getLogger(Logger.ROOT_LOGGER_NAME);
-
-      OpenTelemetryAppender appender = new OpenTelemetryAppender();
-      appender.start();
-      logger.addAppender(appender);
+    // org.springframework.boot.context.logging.LoggingApplicationListener
+    // initializes logging
+    ) {
+      Optional<OpenTelemetryAppender> existingOpenTelemetryAppender = findOpenTelemetryAppender();
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent =
+          (ApplicationEnvironmentPreparedEvent) event;
+      if (existingOpenTelemetryAppender.isPresent()) {
+        reInitializeOpenTelemetryAppender(
+            existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
+      } else {
+        addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
+      }
+    }
+  }
+
+  private static void reInitializeOpenTelemetryAppender(
+      Optional<OpenTelemetryAppender> existingOpenTelemetryAppender,
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent) {
+    OpenTelemetryAppender openTelemetryAppender = existingOpenTelemetryAppender.get();
+    openTelemetryAppender.stop(); // The OpenTelemetry appender is stopped and restarted from the
+    // org.springframework.boot.context.logging.LoggingApplicationListener.initialize
+    // method
+    // The OpenTelemetryAppender initializes the LoggingEventMapper in the start() method. So, here
+    // we stop the OpenTelemetry appender before its re-initialization and its restart.
+    initializeOpenTelemetryAppenderFromProperties(
+        applicationEnvironmentPreparedEvent, openTelemetryAppender);
+    openTelemetryAppender.start();
+  }
+
+  private static void addOpenTelemetryAppender(
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent) {
+    ch.qos.logback.classic.Logger logger =
+        (ch.qos.logback.classic.Logger)
+            LoggerFactory.getILoggerFactory().getLogger(Logger.ROOT_LOGGER_NAME);
+    OpenTelemetryAppender openTelemetryAppender = new OpenTelemetryAppender();
+    initializeOpenTelemetryAppenderFromProperties(
+        applicationEnvironmentPreparedEvent, openTelemetryAppender);
+    openTelemetryAppender.start();
+    logger.addAppender(openTelemetryAppender);
+  }
+
+  private static void initializeOpenTelemetryAppenderFromProperties(
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent,
+      OpenTelemetryAppender openTelemetryAppender) {
+
+    Boolean codeAttribute =
```

Comment:
config properties would make this more concise

Reply:
I have implemented in the same way the Spring `LoggingApplicationListener` is implemented, see [for example](https://github.com/spring-projects/spring-boot/blob/ff97fdc203290b55f010a2ee7263a0be34159b3f/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/logging/LoggingApplicationListener.java#L320). Config properties may not work  because Spring application listeners have the possibility to be executed early in the application start up.

Reply:
from meeting: config properties are not available at this point - please add comment

Reply:
comment added